cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreTableManager.java

```diff
@@ -461,7 +468,13 @@ private String generateHivePartitionKey(PartitionKey key) {
       String fieldName = entry.getKey();
       Comparable fieldValue = entry.getValue();
       String quote = fieldValue instanceof String ? "'" : "";
-      builder.append(sep).append(fieldName).append("=").append(quote).append(fieldValue.toString()).append(quote);
+      builder.append(sep);
+      if (shouldEscapeColumns) {
+        builder.append('`').append(fieldName).append('`');
```

Comment:
There is one tricky issue here, if a column name contains a ' character which apparently was allowed prior to Hive 12. That should never happen in CDAP because as far as I know we do not allow ' in field names of a schema.